skbio/sequence/_genetic_code.py

```diff
@@ -8,588 +8,776 @@
+        >>> gc == named_gc
+        True
+
+        """
+        if self.__class__ != other.__class__:
+            return False
+        # convert Protein to str so that metadata is ignored in comparison. we
+        # only care about the sequence data defining the genetic code
+        if str(self._amino_acids) != str(other._amino_acids):
+            return False
+        if str(self._starts) != str(other._starts):
+            return False
+        return True
+
+    def __ne__(self, other):
+        """Determine if the genetic code is not equal to another.
+
+        Genetic codes are not equal if their type, `amino_acids`, or `starts`
+        differ. A genetic code's name (accessed via ``name`` property) does not
+        affect equality.
 
         Parameters
         ----------
-        codon : str
-            codon string
+        other : GeneticCode
+            Genetic code to test for inequality against.
 
         Returns
         -------
         bool
-            ``True`` if codon is a start codon, ``False`` otherwise
-
-        Examples
-        --------
-        >>> from skbio.sequence import GeneticCode
-        >>> sgc = GeneticCode('FFLLSSSSYY**CC*WLLLLPPPPHHQQRRRRIIIMTTTTNNKKSS'
-        ...                   'RRVVVVAAAADDEEGGGG')
-        >>> sgc.is_start('ATG')
-        False
-        >>> sgc.is_start('AAA')
-        False
+            Indicates whether the genetic code is not equal to `other`.
 
         """
-        fixed_codon = codon.upper().replace('U', 'T')
-        return fixed_codon in self.start_codons
+        return not (self == other)
 
-    def is_stop(self, codon):
-        """Checks if codon is a stop codon
+    def translate(self, sequence, reading_frame=1, start='ignore',
+                  stop='ignore'):
+        """Translate RNA sequence into protein sequence.
 
         Parameters
         ----------
-        codon : str
-            codon string
+        sequence : RNA
+            RNA sequence to translate.
+        reading_frame : {1, 2, 3, -1, -2, -3}
+            Reading frame to use in translation. 1, 2, and 3 are forward frames
+            and -1, -2, and -3 are reverse frames. If reverse (negative), will
+            reverse complement the sequence before translation.
+        start : {'ignore', 'require', 'optional'}
+            How to handle start codons:
+
+            * "ignore": translation will start from the beginning of the
+              reading frame, regardless of the presence of a start codon.
+
+            * "require": translation will start at the first start codon in
+              the reading frame, ignoring all prior positions. The first amino
+              acid in the translated sequence will *always* be methionine
+              (M character), even if an alternative start codon was used in
+              translation (this behavior most closely matches the underlying
+              biology). If a start codon does not exist, a ``ValueError`` is
```

Comment:
Maybe:

*(this behavior most closely matches the underlying biology)* to *(this behavior most closely matches the underlying biology since fMet doesn't have a corresponding IUPAC character)*

I think this extra detail is useful as it'll let users figure out what to search if they want to learn more about this.